src/components/views/auth/CaptchaForm.tsx

```diff
@@ -15,55 +15,56 @@ limitations under the License.
 */
 
 import React, { createRef } from 'react';
-import PropTypes from 'prop-types';
 import { _t } from '../../../languageHandler';
 import CountlyAnalytics from "../../../CountlyAnalytics";
 import { replaceableComponent } from "../../../utils/replaceableComponent";
 
 const DIV_ID = 'mx_recaptcha';
 
+interface ICaptchaFormProps {
+    sitePublicKey: string
+    onCaptchaResponse: () => void
+}
+
+interface ICaptchaFormState {
+    errorText: string | null
+
+}
+
 /**
  * A pure UI component which displays a captcha form.
  */
 @replaceableComponent("views.auth.CaptchaForm")
-export default class CaptchaForm extends React.Component {
-    static propTypes = {
-        sitePublicKey: PropTypes.string,
-
-        // called with the captcha response
-        onCaptchaResponse: PropTypes.func,
-    };
-
+export default class CaptchaForm extends React.Component<ICaptchaFormProps, ICaptchaFormState> {
     static defaultProps = {
-        onCaptchaResponse: () => {},
+        onCaptchaResponse: () => { },
     };
 
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            errorText: null,
-        };
+    private _captchaWidgetId: string | null = null;
+    private _recaptchaContainer: React.RefObject<HTMLDivElement> = createRef();
```

Comment:
given `private` these shouldn't be underscored

Reply:
Also `private _recaptchaContainer = createRef<HTMLDivElement>();` is preferred due to brevity